lib/person.rb

```diff
@@ -0,0 +1,18 @@
+class Person
+
+  attr_reader :name, :interests, :supplies
+
+  def initialize(attributes)
+    @name = attributes[:name]
+    @interests = attributes[:interests]
+    @supplies = {}
+  end
+
+  def add_supply(item, amount)
+   supply = @supplies[item] = amount
```

Comment:
You are on the right track with utilizing the existing @supplies hash. The first step is to check to see if the hash has the type of item you want to add already. If it does you want to add the amount to that item in the hash. If it does not, you want to create a new key and set the value to be the amount passed in. You can do this with an if statement.

```suggestion
if @supplies[item]    #check to see if the item exists in the hash
  @supplies[item] += amount    # if it does, add the amount to the existing amount
else                                                       
  @supplies[item] = amount     # if it does not, add the new item to the hash and set the value equal to the amount.
end
```